src/Sparrow.Server/ByteString.cs

```diff
@@ -472,25 +472,30 @@ public readonly Span<T> ToSpan<T>() where T : unmanaged
 
     public sealed unsafe class UnmanagedGlobalSegment : PooledItem
     {
-        public byte* Segment;
+        private byte* _segment;
+        public byte* Segment => _segment;
+
         public readonly int Size;
+
         private readonly NativeMemory.ThreadStats _thread;
 
         public UnmanagedGlobalSegment(int size)
         {
             Size = size;
-            Segment = NativeMemory.AllocateMemory(size, out _thread);
+            _segment = NativeMemory.AllocateMemory(size, out _thread);
             InUse.Raise();
         }
 
         ~UnmanagedGlobalSegment()
         {
             try
             {
-                if (Segment == null)
+                var segment = _segment;
+                if (segment == null)
                     return;
-                NativeMemory.Free(Segment, Size, _thread);
-                Segment = null;
+                _segment = null;
+
+                NativeMemory.Free(segment, Size, _thread);
```

Comment:
Call `Dispose` here, same code

Reply:
Do you mean into an untyped instead of `ObjectDisposedException` right?